scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/simulation/ResumableSimulationDriver.java

```diff
@@ -328,93 +338,67 @@ private void simulateSchedule(final Map<ActivityDirectiveId, ActivityDirective>
    * @return its duration if the activity has been simulated and has finished simulating, an IllegalArgumentException otherwise
    */
   public Optional<Duration> getActivityDuration(ActivityDirectiveId activityDirectiveId){
+    //potential cause of non presence: activity is outside plan bounds
+    if(!plannedDirectiveToTask.containsKey(activityDirectiveId)) return Optional.empty();
     return engine.getTaskDuration(plannedDirectiveToTask.get(activityDirectiveId));
   }
 
+  private Set<ActivityDirectiveId> getSuccessorsToSchedule(final SimulationEngine engine) {
+    final var toSchedule = new HashSet<ActivityDirectiveId>();
+    final var iterator = toCheckForDependencyScheduling.entrySet().iterator();
+    while(iterator.hasNext()){
+      final var taskToCheck = iterator.next();
+      if(engine.isTaskComplete(taskToCheck.getValue())){
+        toSchedule.add(taskToCheck.getKey());
+        iterator.remove();
+      }
+    }
+    return toSchedule;
+  }
```

Comment:
Minor, nonblocking: proposal: this method could call `scheduleActivities` directly instead of doing `toSchedule.add`. Then we don't have to do create a set. Additionally, it would get rid of an unnecessary loop (we already loop over the set `toScheduleNow` in the process of generating it)

Reply:
But we would be looping through it while we would be adding elements to it in `scheduleActivities`

Reply:
...Ah, yeah, forgot about that.